general/business_model.md

```diff
@@ -0,0 +1,56 @@
+
+Almost every single one of these companies follows what is called the [open-core model](https://en.wikipedia.org/wiki/Open_core). 
+
+> Open core is a business model for the monetization of commercially produced open-source software. ...primarily involves offering a "core" or feature-limited version of a software product as free and open-source software, while offering "commercial" versions or add-ons as proprietary software.
+
+It's important to note that while often proprietary means closed source code, it doesn't have to be. You can have your source code open to the world but with a restrictive license that for instance can only be used for academic research.
+
+**Side-note:** as an organization whatever we do, we should never write/implement our own open source licenses, the community has already done a great job of creating and vetting a selection of licenses that are the status quo.
+
+## source{d} business model
+
+### Who do we want to target as customers?
+
+To make sure we stay true to doing what is right for individual developers in the community, we do not want to charge any individual users.
+
+We don't believe startups and SME's make up a large enough market to build a sustainable company in the long term. The amount you can charge to small organizations pales in comparission to what you can charge enterprises (value add * # of developers). At the same time, selling to SME's is almost always a [SaaS](https://en.wikipedia.org/wiki/Software_as_a_service) business.  While there are success stories of this, such as [Travis CI](https://travis-ci.com/) we don't think it's the right route for source{d}. 
+
+**source{d} focuses on enterprises as its customers**
+
+From the point of view of value we can add to an organization, large scale analysis & ML on top of source code shines when there is a large amount of code and a large number of developers.
+
+### Enterprises as customers
+
+Today every large organization has become/is becoming a software company. The number of software developers in what were once considered traditional businesses, such as banks and airlines, is often already in the 1,000s or 10,000s.
+
+There is a growing number of challenges that enterprises face within their [Software Development Life Cycle (SDLC)](https://en.wikipedia.org/wiki/Systems_development_life_cycle) that can start to be tackled with large scale language-agnostic analysis of their source code, and trained ML models.
+
+### Business model meets our technology
+
+The current model that we are exploring is to have single node versions of our technology stack be fully open-source under a permissive licenses such as [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0) but multiple nodes of our stack that allow distributed computing over a large amount of repositories with a large number of concurent users to be under a restrictive license. 
```

Comment:
Really interested to see, how that would be possible in case of a single codebase.

Personally, I do not see a way for this to happen yet, except if a VERY restrictive license (non-OSS?) is used firsthand, so enterprises are incentivized to pay or buy a re-license of the same codebase, under a comercial license that has the desired terms.

AGPL my be not restrictive enough, as it would allow multiple node usage, in case enterprises do not need modify/re-distributer the “derivative work” on top of this codebase.

But even if they do, it still could be free (as a 🍺) for them to use, if they are willing to contribute all the changes upstream.

But of course I’m not a lawyer, thus the excitement to see how this works out.

Reply:
@bzz this is why we're having the discussion now. Since the new reposdb project (we really need more than a code-name for this project) is now underway, it could open up the possibility to separate code bases of the main data source. Having separation of concerns across different code bases will make it significantly easier for the community to understand what is pure OSS and what is restrictive, otherwise we're making things too confusing.

Reply:
I think you need to consult with a lawyer specialized in open source licenses to evaliuate options to reach your goal.
If it' a single code base, the limitation single node vs distributed you want to enforce will not work since it is against the open source definition. https://en.wikipedia.org/wiki/The_Open_Source_Definition https://opensource.org/faq#commercial

You might want to look into what Chef did in 2014 with a mix of commercial and open source components in the product, and a proprietary license for the commercial pieces that limited the number of nodes for free to 25. https://blog.chef.io/2014/09/08/there-is-one-chef-server-and-it-is-open-source/
https://www.chef.io/online-master-agreement-archive-20151101/

But it seems they changed that in 2016 https://blog.chef.io/2016/04/26/changes-to-how-chef-products-handle-licenses/
https://www.chef.io/online-master-agreement/

I don't think you'll be able to have a single open source codebase covering both open source and limited parts, you will need some kind of proprietary offering targeting Enterprise needs with a proprietary license.

Having that part under a proprietary license if not unfriendly to individual developers: without a business model for sourced there will be no corporation funding the development of the open source offering.

Reply:
@chanezon thank you very much for the insights. What is your opinion on keeping proprietarily licensed code public and not in a private repo?